chacha20/Cargo.toml

```
@@ -20,18 +20,27 @@ categories = ["cryptography", "no-std"]
 
 [dependencies]
 cfg-if = "1"
-cipher = "0.4.4"
-
+cipher = { version = "0.4.4", optional = true}
+rand_core = { version = "0.6.4", optional = true, default-features = false, features = ["getrandom"] }
+serde = { version = "1.0", features = ["derive"], optional = true }
+zeroize = { version = "1.6.0", optional = true }
 [target.'cfg(any(target_arch = "x86_64", target_arch = "x86"))'.dependencies]
 cpufeatures = "0.2"
 
 [dev-dependencies]
 cipher = { version = "0.4.4", features = ["dev"] }
 hex-literal = "0.3.3"
+rand_chacha = "0.3.1"
+serde_json = "1.0" # Only to test serde1
+sha2 = "0.10" # For testing fill_bytes()
+sha3 = "0.10" # Also for testing fill_bytes(), but it may be unnecessary
```

I'm guessing the build failures are due to the lack of `resolver = "2"` combined with feature unification and the addition of these dependencies (which seem unnecessary, but I digress).

You can probably fix it with:

```suggestion
sha2 = { "0.10", default-features = false } # For testing fill_bytes()
sha3 = { "0.10", default-features = false } # Also for testing fill_bytes(), but it may be unnecessary
```

---

Good catch. I will remove try that. It was mainly to ensure that switching to pointers for `fill_bytes()` didn't change anything, but it is still probably unnecessary.

---

I still get errors when running that after your suggestions of `resolver = "2"` and adding that part to `sha2` and `sha3`.

I also tried removing those dev dependencies as well as removing the test that depends on them.

---

Built for me with these removed:

```
$ cargo build  --target thumbv7em-none-eabi
warning: virtual workspace defaulting to `resolver = "1"` despite one or more workspace members being on edition 2021 which implies `resolver = "2"`
note: to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest
note: to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
note: for more details see https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions
   Compiling chacha20 v0.9.1 (/private/tmp/stream-ciphers-rng/chacha20)
    Finished dev [optimized + debuginfo] target(s) in 0.05s
```